text/0021-any-and-all-operators.md

```diff
@@ -62,47 +62,56 @@ The new operators provide a built-in way to concisely express these universal an
 
 ## Detailed design
 
-The Cedar syntax, semantics, and type system are all extended to support the new operators. These extensions are backward compatible, and existing policies won't be affected. In particular, we will keep the `.containsAny` and `.containsAll` operators on sets, even though they can be expressed in terms of `any` and `all`.
+The Cedar syntax, semantics, and type system are all extended to support the new operators. These extensions are backward compatible, and existing policies won't be affected. The design of the operators is inspired by the ForAllValues and ForAnyValue operators in the IAM language.
 
 ### Extending the syntax
 
 This RFC proposes extending the Cedar grammar as follows:
 
 ```
 Access    :=
-  '.' QUANTIFIER '(' Predicate ')'
-  | '.' IDENT ['(' [ExprList] ')']
+  '.' IDENT ['(' [ExprList] ')']
   | '[' STR ']'
+  | '[' QUANTIFIER Predicate ']'
 QUANTIFIER := 'all' | 'any'
-Predicate  := ...
+Predicate  :=
+  INTCMP Expr |
+  'like' PAT |
+  IDENT '(' [ExprList] ')'
+INTCMP := '<' | '<=' | '>' | '>='
 ```
 
-The `Predicate` expression differs from `Expr` in two ways:
-
-1. Predicates can use the keyword `it`. A `Predicate` defines the body of function with a single parameter, and we use `it` as the parameter name. This is similar to [Kotlin's `it` keyword](https://kotlinlang.org/docs/lambdas.html#it-implicit-name-of-a-single-parameter).
+The extended grammar supports applying functions, integer comparisons, and the `like` operator to sets of values.
 
-2. Predicates cannot contain nested `any` or `all` expressions. We disallow nested quantifiers for both performance and analyzability reasons. Nested quantifiers allow writing expressions that run in $O(2^n)$ time, where $n$ is the number of nested quantifiers. For example:
+For example:
 
 ```
-[0,1].all(
-  [0,1].all(
-    ...
-      [0,1].all(true)...))
+context.ips [any isLoopBack()]
+context.ips [all isInRange(ip("192.168.0.1/24"))]
+resource.tags [any like "private*"]
+resource.scores [all greaterThan(principal.preference)]
 ```
 
-Because nested quantifiers are prohibited, there is no need to support predicates with arbitrary parameter names. The `it` keyword sufficiently extends the grammar.
+The grammar doesn't support ooperating on sets of entities (e.g., using `in`) or sets of records (e.g., using `has`).  It also doesn't support combining multiple operators in a single `any`/`all` expression. Boolean combinations need to be expressed separately, outside of `any`/`all`.
```

Comment:
Can we clarify the comment
 > doesn't support operating on sets of entities (e.g., using `in`)
 
 Are sets of entities unsupported entirely, or is it only `in` that is unsupported?  For instance, if there were an extension function that took an entity as an input, could we use that with `any`/`all`?

Reply:
Will clarify: `in` is the only problematic operator we have right now.  Others, such as `is`, are fine.